hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/bucket/BucketCache.java

```diff
@@ -216,9 +216,7 @@
   });
 
   /** Statistics thread schedule pool (for heavy debugging, could remove) */
-  private transient final ScheduledExecutorService scheduleThreadPool =
-    Executors.newScheduledThreadPool(1,
-      new ThreadFactoryBuilder().setNameFormat("BucketCacheStatsExecutor").setDaemon(true).build());
+  private transient ScheduledExecutorService scheduleThreadPool;
```

Comment:
It could still be final?

Reply:
My thought is if we turn the log off according to configuration, it should be better if we just not create such an object. If keep it as final, there will be a redundant scheduleThreadPool object on the heap. So that I make it not final here and create the object when we initialise the bucketcache

Reply:
Fixed in the latest commit.